src/Raven.Server/Config/RavenConfiguration.cs

```diff
@@ -218,11 +220,25 @@ public RavenConfiguration Initialize()
             return this;
         }
 
+        public static void AssertPostInitCanUseCoraxFeature(RavenConfiguration configuration)
+        {
+            if (configuration.Indexing.AutoIndexingEngineType == SearchEngineType.Corax || configuration.Indexing.StaticIndexingEngineType == SearchEngineType.Corax)
+                AssertCanUseCoraxFeature(configuration);
+        }
+        
+        public static void AssertCanUseCoraxFeature(RavenConfiguration configuration)
+        {
+            var featureGuardian = new FeatureGuardian(configuration);
+            featureGuardian.Assert(Feature.Corax, () => $"To use {nameof(Corax)} search engine you have set {nameof(CoreConfiguration.FeaturesAvailability)} as {FeaturesAvailability.Experimental}.");
+        }
+        
         public void PostInit()
         {
             if (ResourceType != ResourceType.Server)
                 return;
 
+            AssertPostInitCanUseCoraxFeature(this);
```

Comment:
Why not doing that in the first line so also for databases? Someone can set define Corax in the database settings - we need to validate that experimental flag it set